File: backend/catalog-service/src/main/java/me/doruk/catalogservice/response/EventResponse.java
package me.doruk.catalogservice.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import me.doruk.catalogservice.entity.Venue;

import java.math.BigDecimal;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EventResponse {
  private Long eventId;
  private String name;
  private int capacity;
  private Venue venue;
  private BigDecimal ticketPrice;
  private String eventDate;
  private String description;
  private String imageUrl;
}

File: backend/CatalogService/Responses/EventResponse.cs
using CatalogService.Entities;


namespace CatalogService.Responses;

public class EventResponse(
    long id, string name, int capacity, Venue venue, decimal ticketPrice, DateTime eventDate, string description, string imageUrl)
{
    public long Id { get; set; } = id;
    public string Name { get; set; } = name;
    public int Capacity { get; set; } = capacity;
    public Venue Venue { get; set; } = venue;
    public decimal TicketPrice { get; set; } = ticketPrice;
    public DateTime EventDate { get; set; } = eventDate;
    public string Description { get; set; } = description;
    public string ImageUrl { get; set; } = imageUrl;
}
